test/framework/src/main/java/org/elasticsearch/snapshots/mockstore/MockRepository.java

```diff
@@ -111,8 +111,15 @@ public long getFailureCount() {
 
     private volatile boolean blockOnDeleteIndexN;
 
-    /** Allows blocking on writing the index-N blob; this is a way to enforce blocking the
-     *  finalization of a snapshot, while permitting other IO operations to proceed unblocked. */
+    /**
+     * Allows blocking on writing the index-N blob and subsequently failing it on unblock.
+     * This is a way to enforce blocking the  finalization of a snapshot, while permitting other IO operations to proceed unblocked.
+     */
+    private volatile boolean blockAndFailOnWriteIndexFile;
+
+    /**
+     * Same as {@link #blockAndFailOnWriteIndexFile} but proceeds without error after unblock.
+     */
     private volatile boolean blockOnWriteIndexFile;
```

Comment:
Not having this was a serious gap in our testing unfortunately. We've had various test failures that were essentially the result of a race with physically writing the next index generation while the repo wasn't yet in a consistent state. The new tests using this significantly up the coverage of this scenario (at the annoying cost of making this class yet again more complex).

Reply:
It worth it IMO